src/server/cluster/cluster_defs.h

```diff
@@ -24,45 +21,85 @@ struct SlotRange {
   SlotId start = 0;
   SlotId end = 0;
 
-  bool operator==(const SlotRange& r) const {
+  bool operator==(const SlotRange& r) const noexcept {
```

Comment:
I see you added many `noexcept` in this area, why though? Generally we don't use exceptions in Dragonfly (except in very limited use cases)

Reply:
we don't use exception but support them, it increases our binary file and reduce performance. So I decided to add noexcept. it should improve performance and reduce binary file

Reply:
It does add noise to the code, and I doubt if you'll be able to measure any difference in runtime with and without this.
My take is: either we do it everywhere, or nowhere. Just adding them in this file isn't helpful and even confusing...

Reply:
I'm going to add it everywhere where I will change the code

Reply:
I wrote that in the Dataplane chat some time ago and nobody answered with no)

Reply:
I don't like this, and the fact that we can't prove that we don't throw (like that an internal function call will run out of memory) doesn't make it better
compilers don't enforce this optimally to my understanding
I'd remove all of these and have a conversation within the team
But I won't insist...